packages/grid/_modules_/grid/components/toolbar/GridToolbarExport.tsx

```diff
@@ -23,116 +14,72 @@ interface GridExportDisplayOptions {
   csvOptions?: GridCsvExportOptions & GridExportDisplayOptions;
   printOptions?: GridPrintExportOptions & GridExportDisplayOptions;
 }
 
-const GridToolbarExport = React.forwardRef<HTMLButtonElement, GridToolbarExportProps>(
-  function GridToolbarExport(props, ref) {
-    const { csvOptions, printOptions, onClick, ...other } = props;
-    const apiRef = useGridApiContext();
-    const rootProps = useGridRootProps();
-    const buttonId = useId();
-    const menuId = useId();
+export const GridCSVExportMenuItem = (props: GridCSVExportMenuItemProps) => {
+  const apiRef = useGridApiContext();
+  const { hideMenu, options } = props;
 
-    const [open, setOpen] = React.useState(false);
-    const buttonRef = React.useRef<HTMLButtonElement>(null);
-    const handleRef = useForkRef(ref, buttonRef);
+  if (options?.disableToolbarButton) {
+    return null;
+  }
+  return (
+    <MenuItem
+      onClick={() => {
+        apiRef.current.exportDataAsCsv(options);
+        hideMenu?.();
+      }}
+    >
+      {apiRef.current.getLocaleText('toolbarExportCSV')}
+    </MenuItem>
+  );
+};
 
-    const exportOptions: GridExportMenuItemsParams[] = [
-      {
-        label: apiRef.current.getLocaleText('toolbarExportCSV'),
-        format: 'csv',
-        formatOptions: csvOptions,
-      },
-      {
-        label: apiRef.current.getLocaleText('toolbarExportPrint'),
-        format: 'print',
-        formatOptions: printOptions,
-      },
-    ];
+export const GridPrintExportMenuItem = (props: GridPrintExportMenuItemProps) => {
+  const apiRef = useGridApiContext();
+  const { hideMenu, options } = props;
 
-    const handleMenuOpen = (event) => {
-      setOpen(true);
-      onClick?.(event);
-    };
-    const handleMenuClose = () => setOpen(false);
-    const handleExport = (option: GridExportMenuItemsParams) => () => {
-      switch (option.format) {
-        case 'csv':
-          apiRef.current.exportDataAsCsv(option.formatOptions);
-          break;
-        case 'print':
-          apiRef.current.exportDataAsPrint(option.formatOptions);
-          break;
-        default:
-          break;
-      }
+  if (options?.disableToolbarButton) {
+    return null;
+  }
+  return (
+    <MenuItem
+      onClick={() => {
+        apiRef.current.exportDataAsPrint(options);
+        hideMenu?.();
+      }}
+    >
+      {apiRef.current.getLocaleText('toolbarExportPrint')}
+    </MenuItem>
+  );
+};
 
-      setOpen(false);
-    };
-
-    const handleListKeyDown = (event: React.KeyboardEvent) => {
-      if (isTabKey(event.key)) {
-        event.preventDefault();
-      }
-      if (isHideMenuKey(event.key)) {
-        handleMenuClose();
-      }
-    };
+const GridToolbarExport = React.forwardRef<HTMLButtonElement, GridToolbarExportProps>(
+  function GridToolbarExport(props, ref) {
+    const { csvOptions = {}, printOptions = {}, ...other } = props;
 
     if (csvOptions?.disableToolbarButton && printOptions?.disableToolbarButton) {
```

Comment:
This condition is no longer correct

Reply:
It is correct, because `<GridToolbarExport />` is the component you can customize only by passing props.

If a dev wants to use a custom export format, She must define its own ToolbarExport, on the same idea that you can create your own Toolbar with custom elements by using `GridToolbarContainer` 

```jsx
import {GridToolbarExportContainer, GridCSVExportMenuItem} from '@mui/x-data-grid-pro'

const MyToolbarExport = (props) => (
	<GridToolbarExportContainer {...props}>
		<GridCSVExportMenuItem options={csvOptions} />
		<MyOwnPrintExportMenuItem />
	</GridToolbarExportContainer>
)
```

You can incorporate this new button in you toolbar as follow:

```jsx

const CustomToolbar = () => (
    <GridToolbarContainer>
      <MyToolbarExport />
    </GridToolbarContainer>
)

...

<DataGrid
	...
        components={{
          Toolbar: CustomToolbar,
        }}
      />
```